homework2/task01/project1/task1.py

```diff
@@ -0,0 +1,78 @@
+"""
+Given a file containing text. Complete using only default collections.
+
+1) Find 10 longest words consisting from largest amount of unique symbols
+2) Find rarest symbol for document
+3) Count every punctuation char
+4) Count every non ascii char
+5) Find most common non ascii char for document
+"""
+import re
+from collections import Counter
+from typing import List
+
+
+def file_presettings(file_path: str) -> str:
+    r"""
+    Return unpacked file, encode in readable format.
+
+    First sub() deletes '\n' between lines. Second sub() remove punctuation.
+    """
+    with open(file_path, "r") as f:
+        text = f.read()
+        a = text.encode().decode("unicode_escape")
+        filtered_text = re.sub(r"[\n]", " ", a)
+        return re.sub(r"[^\w]", " ", filtered_text)
+
+
+def get_longest_diverse_words(file_path: str) -> List[str]:
+    """Return 10 longest diverse words."""
+    text = file_presettings(file_path)
+    all_words = set(text.split())
+    result = []
+    result.extend(all_words)
+    result.sort(key=lambda i: (len("".join(set(i))), i[0]), reverse=True)
```

Comment:
Please replace lambda with the function.

Reply:
Done